[YHIOS-63]: Add YHCustomAlert component

foxsetDev opened the pull request:
- Документация по Компоненту -> [WikiYandex](https://wiki.yandex.ru/homepage/ce33d84eb842/dizajjn-sistemacommonui/komponent-yhcustomalert/)

https://github.com/user-attachments/assets/bb3c0c39-90d5-479b-b257-a0c0ad0c4548

foxsetDev committed:
YHIOS-63: Add Alert Icons to CommonUI assets

Добавил Иконки в Ассеты

foxsetDev committed:
Add YHCustomAlert to CommonUI

foxsetDev committed:
Merge branch 'develop' into YHIOS-63-Add-YHCustomAlert-component

foxsetDev committed:
Cherry-picked from branch YHIOS-34

Исправления Коллекций

foxsetDev committed:
Fix Frame (bodyHeight)

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+    var buttonTitle: String
+    
+    var onButtonTap: () -> Void
+    var onClose: () -> Void
```
нужны модификаторы доступа

foxsetDev replied:
done

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+    var body: some View {
+        ZStack {
+            Color.black300
+                .ignoresSafeArea()
```
Перекроет фон, по дизайну предыдущий View виден

foxsetDev replied:
done

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+                    .padding(.top, Constants.iconMessagePadding)
+                
+                Text(title)
+                    .font(.title3)
```
шрифт не по дизайну

foxsetDev replied:
done

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+                    .font(.title3)
+                    .multilineTextAlignment(.center)
+                    .padding(.horizontal)
+                    .lineLimit(2)
```
для чего лимит?

foxsetDev replied:
@Fedor-Zadorozhnyi посмотрел в Фигме
у нас пока - что в двух местах где может заюзаться Алерт : Вход и Рега и Тренажор

Лимита достаточно было бы

foxsetDev replied:
done

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+                Button(action: onButtonTap) {
+                    Text(buttonTitle)
+                        .font(Constants.buttonTypography)
+                        .foregroundStyle(.purple)
```
цвет не по дизайну

foxsetDev replied:
done

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+    }
+    
+    private var closeButton: some View {
+        Button(action: onClose) {
```
как будто обычного dismiss достаточно

foxsetDev replied:
@Fedor-Zadorozhnyi 
ну `dismiss` он только закроет окно и логику не накинуть будет совсем + он нужен если системный был Алерт

а у нас раз Кастомный Алерт
 можно будет и навигацию какую-нибудь накинуть после закрытия алерта
чисто из этих соображений 
 
 что думаешь?

Fedor-Zadorozhnyi replied:
[dismiss](https://developer.apple.com/documentation/swiftui/environmentvalues/dismiss) в принципе завершает текущую презентацию. А вот если в задумке дизайнера и есть какая-то специфическая навигация после закрытия представления, то в таком случае подходит и твой вариант

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+    enum Constants {
+        static let bodyWidth: CGFloat = 346
+        static let bodyHeight: CGFloat = 358
+        static let cornetRadius: CGFloat = 20
```
опечатка в corner

foxsetDev replied:
done

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+        static let iconMessagePadding: CGFloat = 28
+        
+        static let massageHorizontalPadding: CGFloat = 35
+        static let buttonPadding: CGFloat = 50
```
а точно ли 50? как будто меньше

foxsetDev replied:
@Fedor-Zadorozhnyi точно 50

Fedor-Zadorozhnyi replied:
я там 32 вижу, странно

foxsetDev replied:
мы про разное скорее всего
sizeButton - 32
padding от `massage` -  50

Fedor-Zadorozhnyi replied:
вот контекст:

```
Button(action: onButtonTap) {
                    Text(buttonTitle)
                        .font(Constants.buttonTypography)
                        .foregroundStyle(.purple)
                        .frame(maxWidth: .infinity)
                }
                .padding(.bottom, Constants.buttonPadding)
```

в дизайне:

1. высота кнопки являющейся текстом 17
2. расстояние до top Pop-up 297
3. размер всего Pop-up 346
исходя из этого padding для bottom кнопки 346 - 297 - 17 = 32

foxsetDev replied:
@Fedor-Zadorozhnyi 
контекст ясен) просто перфитПиксель легко проверить следующими танцами ->


зажимаю option и навожу на контейнер с Текстом там расстояние" 50" до контейнера с Баттоном
@unknownkirya 
или так не делаем? 
не верны действия?

https://github.com/user-attachments/assets/f542b539-9408-4f31-bf29-eeafc2bddaec

unknownkirya replied:
все так, оставляем 50

unknownkirya replied:
опечатка в `massage`

Fedor-Zadorozhnyi replied:
Bottom это низ же)
<img width="335" height="346" alt="Снимок экрана 2025-09-22 в 11 52 46" src="https://github.com/user-attachments/assets/c213cef8-9b7c-4eb4-b172-b94e13834968" />

foxsetDev replied:
Это уже 82 будет) а не 32😀

Fedor-Zadorozhnyi replied:
ты задаешь расстояние для кнопки, относительно контейнера, от внешней  нижней части контейнера, чтобы задать расстояние между кнопкой и текстом алерта 50 - нужно использовать другой паддинг(top)

Fedor-Zadorozhnyi replied:
но для того, чтобы не было конфликта - нужно фрейм контейнера убрать, тот самый который ты жестко фиксировал

foxsetDev replied:
Добренько 
Вечером гляну🤝🤌

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+                    Text(buttonTitle)
+                        .font(Constants.buttonTypography)
+                        .foregroundStyle(.purple)
+                        .frame(maxWidth: .infinity)
```
[Текстовое представление всегда занимает ровно столько места, сколько необходимо для отображения его содержимого](https://developer.apple.com/documentation/swiftui/text#:~:text=A%20text%20view%20always%20uses%20exactly%20the%20amount%20of%20space%20it%20needs%20to%20display%20its%20rendered%20contents)

foxsetDev replied:
@Fedor-Zadorozhnyi 

хочешь сказать .инфинити - только для кнопок актуально?
если текст длинным будет, всеровно не спасет? так?

Fedor-Zadorozhnyi replied:
По сути, он по умолчанию инфинити, тут нужно исходить из того, что ты хотел получить, я бы ограничил ширину, чтобы в случае чего текст красиво отображался

unknownkirya replied:
лучше убрать инфинити тут

foxsetDev replied:
done

Fedor-Zadorozhnyi commented:
В гитфлоу есть структура веток, сейчас не по структуре
Названия коммитов тоже не по гитфлоу

Fedor-Zadorozhnyi commented on CommonUI/CommonUI/Sources/YHCustomAlert.swift:
```diff
@@ -0,0 +1,101 @@
+                }
+                .padding(.bottom, Constants.buttonPadding)
+            }
+            .frame(
+                width: Constants.bodyWidth,
+                height: Constants.bodyHeight
+            )
```
Жесткий фрейм, может стоит обойтись паддингами? тогда и lineLimit и Spacer будут не нужны

unknownkirya replied:
да, Федя прав, если мы в будущем например айпад начнем поддерживать, мы офигеем это править)

unknownkirya commented:
Так, текст и картинка Алерта у нас слишком кастомные - под одно состояние, предлагаю сделать так:
1) добавить состояние в компонент - default(базовый Алерт без картинки, с текстом и с одной кнопкой), emailConfirmationAlert(который сейчас у тебя), custom(с возможностью добавлять свою картинку,  текст и настраивать количество кнопок(1 или 2)) - это все делаем через enum
2) делаем инит компонента, передаем его настройку в init
3) добавляем скрины новых состояний к PR

foxsetDev committed:
Merge branch 'develop' into YHIOS-63-Add-YHCustomAlert-component

foxsetDev committed:
cherry/YHIOS-72: The new fonts have been added

ЧерриПик с YHIOS-72

foxsetDev committed:
[YHIOS-63]: add custom case to YHCustomAlert

- Добавлены три состояния Alert`a ( custom, emailConfirmationAlert, default)
- Были удалены фреймы, на Padding все переверстал
- Исправлен Шрифт в Тайтле
-

foxsetDev commented:
- **YHcustomAlertTwoButton**
 ![YHcustomAlertTwoButton](https://github.com/user-attachments/assets/46d3a68e-7f5e-4f54-88bf-b05f7c86e747)

- **YHcustomAlertEmail**
 ![YHcustomAlertEmail](https://github.com/user-attachments/assets/d94e2127-f564-4460-b681-1f411a89c1d9)

- **YHcustomAlertDefault**
 ![YHcustomAlertDefault](https://github.com/user-attachments/assets/9d441af4-ff06-4278-a552-6f41955efb10)

unknownkirya commented:
approved
